lib/internal/promiseCallback.js

```diff
@@ -0,0 +1,19 @@
+const PROMISE_SYMBOL = Symbol('promiseCallback')
+
+function promiseCallback () {
```

Comment:
This is the older strategy.  It's a bit more clunky, but doesn't add another frame to the call stack for each method.

Reply:
I had to use this in a couple places where the `awaitify` wrapper couldn't work  (usually due to optional arguments).